packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemNextModule.kt

```diff
@@ -0,0 +1,80 @@
+package expo.modules.filesystem.next
+
+import expo.modules.kotlin.apifeatures.EitherType
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+import expo.modules.kotlin.typedarray.TypedArray
+import expo.modules.kotlin.types.Either
+import java.io.File
+import java.net.URI
+
+class FileSystemNextModule : Module() {
+
+  @OptIn(EitherType::class)
+  override fun definition() = ModuleDefinition {
+    Name("FileSystemNext")
+
+    Class(FileSystemNextFile::class) {
+      Constructor { path: String ->
```

Comment:
Afaik we already have a converter for `URI` which could handle some edge cases and do "string -> uri" conversion in the same way as other modules do.